### ISSUE TITLE
Invalid account id should return a proper JSON error HTTP/400

### ISSUE THREAD

**Issue by frol** [telezhnaya/near_enhanced_api#16]
<img width="522" alt="image" src="https://user-images.githubusercontent.com/304265/178987309-875fc1af-2c28-431a-b762-9062ac3e2f16.png">

**Comment by telezhnaya** [comment]
This error appears because we've made strictly typed query params. We can't parse AccountId here, that's why we have 404, we don't have the endpoint to handle this.
It can be fixed by changing all the query params to simple types (AccountId -> String) and adding the cast&check on our side

@pisomanik, if you are still ready to contribute, do you want to take this one? 🙂

**Comment by telezhnaya** [comment]
Even better solution is to leave AccountId types everywhere and find a way to change the error handling logic, it's a good idea to check whether it's possible

**Comment by munanadi** [comment]
Hello. @telezhnaya 

TBH I can only read rust and make sense of it currently. I think I can work around looking at existing code though.
 
I know it's a lot to ask for, but can you help me with the setup to locally run queries and set up my local dev.
That would be great. If you can please ping me on discord @spoiler#3664

~~Else I will just look around and try to get as far as I can.~~

I believe the line you're talking about is here?
https://github.com/near/near-enhanced-api-server/blob/6c535c2ab6d9d6f087bdf46a187b54b2b67cec76/src/modules/mod.rs#L6-L20

**Comment by telezhnaya** [comment]
@pisomanik sorry I don't have the capacity to help you much. If you are new to Rust, it would be painful for you to start from this task. I'll handle it myself later

**Comment by frol** [comment]
> Even better solution is to leave AccountId types everywhere and find a way to change the error handling logic, it's a good idea to check whether it's possible

Yes, this is the ideal solution we should aim to since we will need to make sure all the handlers return JSON outputs anyway, and it is not great to downscale our validation to Strings instead of proper types.

@pisomanik This task seems to be more advanced than you can currently take. Let's see what else we can offer as an easier starting point for contributions. Thanks for your interest and willingness to contribute!

**Comment by 0xnikter** [comment]
@telezhnaya looks like there are some problems with a paperclip dep. I added actix-web-validator dep and a custom error handling: https://github.com/near/near-enhanced-api-server/compare/main...neeeekitos:near-enhanced-api-server:main

The compiler doesn't find functions update_parameter, update_security, update_security_definitions, update_definitions in paperclip. But normally actix-web-validator support has been added in [paperclip v4.0.0](https://github.com/paperclip-rs/paperclip/pull/403/files). Do have any ideas why the compiler doesn't find these functions?
